locator/network_topology_strategy.cc

```diff
@@ -361,6 +361,114 @@ future<tablet_map> network_topology_strategy::allocate_tablets_for_new_table(sch
     co_return tablets;
 }
 
+#define kqdbg(fmt_, ...) fmt::print(stderr, fmt_ "\n", ##__VA_ARGS__)
+future<tablet_map>
+network_topology_strategy::reallocate_tablets_for_new_rf(schema_ptr s, token_metadata_ptr tm, std::unordered_map<sstring, size_t> new_dc_rep_factor) const {
```

Comment:
If we return a new tablet_map, then it should reflect replica set change as transitions (`set_tablet_transition_info()`) or build a `migration_plan` to be executed (like in tablet_allocator). That's because replica set changes will need to go through a transition state machine.

See how tablet_migration_info from migration_plan is converted into transition in topology_coordinator::generate_migration_update. It's converted directly to mutation of system.tablets, similar fields would have to be set in tablet_transition_info which later gets converted to mutation.

Reply:
See apply_plan_as_in_progress() from the unit test, it reflects migration plan in tablet_map as transition.

Reply:
I thought that I can use the new replicas and pass them directly to the tablet mutation, as in here https://github.com/scylladb/scylladb/pull/16723/files#diff-7367612c80d45bdd40bc28c7b3edc3a163a53cb802b839c3eae1112eb0611f28R706, no?
Something like:
```
updates.emplace_back(replica::tablet_mutation_builder(guard.write_timestamp(), table->id())
    .set_replicas(last_token, tablet_info.replicas)
    ...

Reply:
Maybe I can just copy-paste these changes into the general draft so it's easier to read the code?

Reply:
> I thought that I can use the new replicas and pass them directly to the tablet mutation, as in here

That works too.